### ISSUE TITLE
[BottomNavigation] Horizontal layout should be triggered on .Regular horizontalSizeClass, not landscape oreintation

### ISSUE THREAD

**Issue by romoore** [material-components/material-components-ios#3044]
### Overview
MDCBottomNavigation is currently checking for device orientation == landscape when potentially laying-out labels in a horizontal fashion.  Instead, it should be using the current trait collection and inspecting the horizontal size class.  This will allow split-screen multi-tasking in .Compact environments to correctly use stacked (vertical) label layouts.

### Reproduction steps
1. Open the catalog in an iPad
1. Open BottomNavigation &rarr; Demo
1. Put the device in landscape with split-screen multitasking enabled

Tabs should have the labels below the icons.

### Version number
46.0.0

### Operating system and device
iPad/iOS 10+

### Screenshot
![img_0033](https://user-images.githubusercontent.com/1753199/36997005-417c4e9a-2086-11e8-9da6-d42ef35745a8.PNG)
